src/coreclr/jit/gentree.h

```diff
@@ -8313,6 +8314,41 @@ inline bool GenTree::IsVectorZero() const
     return IsCnsVec() && AsVecCon()->IsZero();
 }
 
+//-------------------------------------------------------------------
+// IsVectorCreate: returns true if this node is the creation of a vector.
+//                 Does not include "Unsafe" method calls.
+//
+// Returns:
+//     True if this node is the creation of a vector
+//
+inline bool GenTree::IsVectorCreate() const
+{
+#ifdef FEATURE_HW_INTRINSICS
+    if (OperIs(GT_HWINTRINSIC))
+    {
+        switch (AsHWIntrinsic()->GetHWIntrinsicId())
+        {
+            case NI_Vector128_Create:
+#if defined(TARGET_XARCH)
+            case NI_Vector256_Create:
+#elif defined(TARGET_ARMARCH)
+            case NI_Vector64_Create:
+#endif
+                return true;
+
+            default:
+                return false;
+        }
+    }
+#endif // FEATURE_HW_INTRINSICS
+
+#ifdef FEATURE_SIMD
+    return OperIs(GT_SIMD);
```

Comment:
You probably want to check the ID is `SimdIntrinsicInit`, `...InitArray`, `...InitArrayX`, `...InitN`, or `...InitFixed`: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/simdintrinsiclist.h

Reply:
-- Looking at how this is used below, you probably want `Init` and `InitN`

`InitArray` and `InitArrayX` take a primitive array and so won't qualify the constant checks

Reply:
`InitFixed` might be interesting in that `new Vector3(new Vector2(5, 4), 3)` could qualify for transform to a `GT_CNS_VEC { 5, 4, 3 }`. But I expect that's not important for reducing regressions.

Reply:
I'm just going to look at `Init` and `InitN`.